src/arch/riscv/rv64/src/bootblock.S

```diff
@@ -29,23 +29,10 @@ _boot:
 spin:
 	bgt a0, x0, spin
 
-	# initialize stack point for each hart
-	# and the stack must be page-aligned.
-	# 0xDEADBEEF used to check stack overflow
-	#csrr a0, mhartid
-	#la   t0, _stack
-	#slli t1, a0, 12 # PageShift
-	#add  t0, t0, t1
-	#li   t1, 0xDEADBEEF
-	#sd t1, 0(t0)
-	#li   t1, 4096 - 88 # RISCV_PGSIZE - HLS_SIZE
-	#add  sp, t0, t1
-
-	# TODO: SOC-specific
-	# Set stack pointer into CAR (L2-LIM).
-1:	auipc t0,     %pcrel_hi(_stack_ptr)
-	lw sp, %pcrel_lo(1b)(t0)
-	#lw sp, 0(t0)
+	# Set the stack pointer
+	# 0xDEADBEEF is used to check stack overflow
+1:     auipc t0,     %pcrel_hi(_stack_ptr)
+       lwu sp, %pcrel_lo(1b)(t0)
```

Comment:
quick question: won't we need the commented out code at some point? How will we do sp setup for the harts?

Reply:
We probably will, but I find having commented out code like this difficult to read. There is also no information on if the code ever worked so I don't think it's very useful to keep around.

Reply:
Ah, this code totally works, it's right from working coreboot on several RISCV platforms.
I'm ok removing it as we have coreboot for backup, but .... it works. The main hangup was gnu assembly to rust assembly conversion.

Reply:
Ah ok.

In which case when re-adding it we should take it from coreboot again. Right now it's just rotting away as no one is using it.